### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by avnerbraverman**

### CODE REVIEW

File: .prettierrc.js [hunk 1]
```diff
@@ -2,6 +2,6 @@ module.exports = {
   semi: false,
   trailingComma: "all",
   singleQuote: true,
-  printWidth: 100,
+  printWidth: 80,
```
Comment (duckranger) [inline .prettierrc.js:5]: Why deviate from all other connectors here?

File: src/AWSMediaConvertConnector.ts [hunk 2]
```diff
@@ -49,20 +61,27 @@ export class AWSMediaConvertConnector extends BaseAWSConnector {
         endpoint: ep.Endpoints[0].Url,
       })
     }
-    return this.mc!
+    if (!this.mc) {
+      throw new Error('No MeriaConvert')
```
Comment (duckranger) [inline src/AWSMediaConvertConnector.ts:65]: `MediaConvert` ?

File: src/BaseAWSConnector.ts [hunk 3]
```diff
@@ -169,7 +183,9 @@ class AWSIdentity {
           await iam
             .putRolePolicy({
               PolicyDocument: JSON.stringify(policy),
-              PolicyName: `policy_${roleName}_${crypto.randomBytes(4).toString('hex')}`,
+              PolicyName: `policy_${roleName}_${crypto
+                .randomBytes(4)
+                .toString('hex')}`,
```
Comment (duckranger) [inline src/BaseAWSConnector.ts:188]: IMO the change here makes the line less readable :(

File: src/CoreConnector.ts [hunk 4]
```diff
@@ -20,21 +21,23 @@ export class CoreEventManager {
   constructor(private connector: BaseConnector) {}
 
   public addEvent(
-    eventOptions: any,
+    eventOptions: Options,
     handler: CoreEventHandler,
     eventId: string | Record<string, any>,
   ): EventConfiguration {
     const id =
       typeof eventId == 'string'
```
Comment (duckranger) [inline src/CoreConnector.ts:29]: Shouldn't we use `===` ?

File: usecases/create-video-thumbnails-et.js [hunk 5]
```diff
@@ -129,7 +134,9 @@ app.setPersistentStore(new MemoryStoreAdapter())
 
   aet.on({ pipelineId: pipeline.Id }, async (event) => {
     console.log(
-      `Trancoding job ${event.jobId}: ${event.old.Status || 'New'} -> ${event.current.Status}`,
+      `Trancoding job ${event.jobId}: ${event.old.Status || 'New'} -> ${
```
Comment (duckranger) [inline usecases/create-video-thumbnails-et.js:137]: `Transcoding` ?

File: doc/AWSLambdaConnector.md [hunk 6]
```diff
@@ -67,6 +67,8 @@ _Connector actions_:
 
 [createFromFile](#createFromFile) Create a new Lambda function from file
 
+[createInFolder](#createInFolder) Creare a new Lambda function by collecting files in a folder
```
Comment (duckranger) [inline doc/AWSLambdaConnector.md:70]: Create

File: doc/AWSLambdaConnector.md [hunk 7]
```diff
@@ -310,6 +312,40 @@ Create a new Lambda function with the given `functionName` to execute
...
+```js
+const functionInfo = await awsLambdaConnector.createInFolder(
+  'toLowerUpperString',
+  async folder => {
+    await folder.copy('index.js', `${__dirname}/toLowerUpperString.js`)
+  }
+)
+```
+
+Create a new Lambda function with the given `functionName` to by setitng up
```
Comment (duckranger) [inline doc/AWSLambdaConnector.md:338]: `to by setitng up` =>? `by setting up`

File: src/AWSLambdaConnector.ts [hunk 8]
```diff
@@ -303,6 +292,31 @@ export class AWSLambdaConnector extends BaseAWSConnector {
...
+    functionName: string,
+    folderHandler: (folder: Folder) => Promise<void>,
+    options: Options = {},
+  ): Promise<any> {
+    const folder = new Folder(
+      `reshuffle-awslambdaconnector-${crypto.randomBytes(8).toString('hex')}`,
+    )
+
+    try {
+      console.log(`${functionName}: Building code folder`)
```
Comment (duckranger) [inline src/AWSLambdaConnector.ts:305]: This and next `console.log` statements could possibly be `console.debug`

**Review by duckranger — CHANGES_REQUESTED** [review]

File: .prettierrc.js (hunk 1 above)
Comment (avnerbraverman) [reply .prettierrc.js:5]: Fixed

File: src/AWSMediaConvertConnector.ts (hunk 2 above)
Comment (avnerbraverman) [reply src/AWSMediaConvertConnector.ts:65]: Fixed

File: src/BaseAWSConnector.ts (hunk 3 above)
Comment (avnerbraverman) [reply src/BaseAWSConnector.ts:188]: Fixed by col width

File: src/CoreConnector.ts (hunk 4 above)
Comment (avnerbraverman) [reply src/CoreConnector.ts:29]: I think this was a typo in the original code. Also fixed in CoreConnector.onStart

File: usecases/create-video-thumbnails-et.js (hunk 5 above)
Comment (avnerbraverman) [reply usecases/create-video-thumbnails-et.js:137]: Fixed

File: doc/AWSLambdaConnector.md (hunk 6 above)
Comment (avnerbraverman) [reply doc/AWSLambdaConnector.md:70]: Fixed

File: doc/AWSLambdaConnector.md (hunk 7 above)
Comment (avnerbraverman) [reply doc/AWSLambdaConnector.md:338]: Fixed

File: src/AWSLambdaConnector.ts (hunk 8 above)
Comment (avnerbraverman) [reply src/AWSLambdaConnector.ts:305]: Fixed

**Comment by avnerbraverman** [comment]
Thanks @duckranger. Pushed fixes. Please take another look

File: doc/AWSLambdaConnector.md [hunk 9]
```diff
@@ -335,7 +335,7 @@ const functionInfo = await awsLambdaConnector.createInFolder(
 )
 ```
 
-Create a new Lambda function with the given `functionName` to by setitng up
+Create a new Lambda function with the given `functionName` to by setting up
```
Comment (duckranger) [inline doc/AWSLambdaConnector.md:348]: Still extra `to` :(

File: doc/AWSLambdaConnector.md [hunk 10]
```diff
@@ -63,7 +63,7 @@ _Connector actions_:
 
 [createFromBuffer](#createFromBuffer) Create a new Lambda function from buffer
 
-[createFromCode](#createFromCode) Create a new Lambda function from code
+[createFromCode](#createFromCode) Create a new Lambda function from code string
```
Comment (duckranger) [inline doc/AWSLambdaConnector.md:66]: Nit: we are sort of overloading the term 'code' in the various actions. In one meaning - it is 'raw' code (like here) and in the other - it is existing code (like in `updateCode` below).
This is not an error but could get confusing. Especially in `updateCodeFromCode`...

As an aside -why do we add these wrapper functions

**Review by duckranger — APPROVED** [review]
Approved - but please fix the extra `to`

File: doc/AWSLambdaConnector.md (hunk 9 above)
Comment (avnerbraverman) [reply doc/AWSLambdaConnector.md:348]: Fixed

File: doc/AWSLambdaConnector.md (hunk 10 above)
Comment (avnerbraverman) [reply doc/AWSLambdaConnector.md:66]: Useful

**Pull request merged**